### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by GlowingScrewdriver**
Tests being added (will update as and when they are done):
* `add-overflow`: Calculate the largest signed integer that can be stored in `int`, and add 1 to it. The result should be the smallest signed integer.
  _Note: this test was ported to use the 32-bit `int` currently in use by llama.lisp, as opposed to 64-bit in upstream Bril_
* `nop`: The no-operation instruction.

### CODE REVIEW

**Comment by GlowingScrewdriver** [comment]
Fixed #10 while porting `add-overflow`.

File: src/backend/llvm.py [hunk 1]
```diff
@@ -209,8 +210,10 @@ def gen_function_prototype(self, fn):
         return func
 
     def declare_var(self, typ, name):
-        """Allocate a pointer using alloc and add it to the symbol table, if it doesn't already exist"""
-        if (name not in self.func_alloca_symtab) or (self.func_alloca_symtab[name].type != typ):
+        """Allocate a pointer using alloca and add it to the symbol table, if it doesn't already exist"""
+        if (name not in self.func_alloca_symtab) or (
+            self.func_alloca_symtab[name].type != typ.as_pointer()
```
Comment (chsasank) [inline src/backend/llvm.py:215]: ah simple fix. This was a bug then

**Comment by chsasank** [comment]
Ignore:
1. [non_linear_control_flow.bril](https://github.com/sampsyo/bril/blob/main/test/interp/core/non_linear_control_flow.bril)
2. main-*-args: dependent of #12 

And document whatever else we're ignoring.

File: src/backend/llvm.py [hunk 2]
```diff
@@ -162,8 +162,9 @@ def gen_comp(instr):
             if "label" in instr:
                 gen_label(instr)
             elif self.builder.block.is_terminated:
-                # Do not codegen for unreachable code
-                instr.op = "dummy"  # This doesn't really do anything
+                pass  # Do not codegen for unreachable code
```
Comment (chsasank) [inline src/backend/llvm.py:165]: what is this for?

Comment (GlowingScrewdriver) [reply src/backend/llvm.py:165]: LLVM requires basic blocks to satisfy 2 constraints:
* Must start with a label
* Must end with a terminating instruction (e.g. unconditional `br`, `ret`)

```llvm
define i32 @"main"()
{
alloca_sIcFeteBBN:
  %"v" = alloca i32
  %"tmp" = alloca i32
  br label %"entry_uvsCZgoUUv"
entry_uvsCZgoUUv:
  store i32 4, i32* %"v"
  br label %"somewhere"
  store i32 2, i32* %"v" ; This needs to be removed, as it is unreachable and doesn't belong in a basic block.
somewhere:
  %".4" = load i32, i32* %"v"
  %"tmp.1" = call i32 @"print"(i32 %".4")
  store i32 %"tmp.1", i32* %"tmp"
  %".6" = load i32, i32* %"v"
  ret i32 %".6"
}

```

**Pull request merged**